File: Java/DecoratorPattern/src/coffee/Espresso.java
package coffee;

/**
 * Author: Utkarsh Dayal @ descifrador17
 * Date: 27-05-2021
 * Time: 04:23 AM
 */
public class Espresso extends Coffee{

    public Espresso(){
        description = "Espresso : Rs "+cost();
    }

    @Override
    public double cost() {
        return 20;
    }
}

File: Kotlin/DecoratorPattern/src/coffee/Espresso.kt
package coffee

/**
 *
 *   Author: Utkarsh Dayal @ descifrador17
 *   Date: 27-05-2021
 *   Time: 04:55 AM
 *
 */
class Espresso : Coffee() {

    init {
        description = "Espresso : Rs ${cost()}"
    }

    override fun cost(): Double {
        return 10.0
    }
}